src/k8s_helpers.rs

```diff
@@ -0,0 +1,30 @@
+use {
+    k8s_openapi::{api::core::v1::Secret, ByteString},
+    kube::api::ObjectMeta,
+    std::{collections::BTreeMap, error::Error, path::PathBuf},
+};
+
+fn create_secret(name: &str, data: BTreeMap<String, ByteString>) -> Secret {
+    Secret {
+        metadata: ObjectMeta {
+            name: Some(name.to_string()),
+            ..Default::default()
+        },
+        data: Some(data),
+        ..Default::default()
+    }
+}
+
+pub fn create_secret_from_files(
+    secret_name: &str,
+    key_files: &[(PathBuf, &str)], //[pathbuf, key type]
+) -> Result<Secret, Box<dyn Error>> {
```

Comment:
So we do have a generic [create_secret()`](https://github.com/anza-xyz/validator-lab/pull/8/files#diff-8c7acc6d7eda3e38ece7000a1feaa0a0c3b72d6c133fe56d871cb16e5ca94865R7-R16) method that is called by `create_secret_from_files`. Is that what you mean? Or you want something different?

Reply:
oh. sorry for the confusion. I mean we have 

https://github.com/anza-xyz/validator-lab/blob/1a22c6c892a8e59f2309923b52834c875e6570dd/src/k8s_helpers.rs#L26

perhaps we could have the caller handle the key name instead of hardcoding it here 🤔 

then we can have this one easier

```rust
pub enum SecretType {
    Value { v: String },
    File { path: PathBuf },
}

pub fn THE_PUB_CREATE_SECRET_FN_NAME (name: &str, hm: HashMap<String, SecretType>) {
    let mut m: BTreeMap<String, String> = BTreeMap::new();
    for (key, value) in &hm {
        match value {
            SecretType::Value { v: _ } => // insert into the btreemap directly
            SecretType::File { path: _ } => // single file version of create_secret_from_files
        }
    }
    create_secret(name, m)
}
```

It's just an idea that came to mind. There's no need to update it unless we have such requests in our code 🙈

Reply:
this is a good idea. i have changed it up in PR: #9 , lmk your thoughts